File: 1436 destination-city/solution.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        set<string> s;
        for(auto &l:paths) s.insert(l[0]);
        for(auto &l:paths) {
            if(s.count(l[1])==0) return l[1];
        }
        return "";
    }
};

File: 1436 destination-city/solution.go
func destCity(paths [][]string) string {
    set := make(map[string]bool)
    for _,v := range paths {
        set[v[0]] = true
    }
    for _,v := range paths {
        if !set[v[1]] {
            return v[1]
        }
    }
    return ""
}
